### PR TITLE
[frd-438] Solana waitForTransaction support

### PR DESCRIPTION
**Pull request by nrjshka**
Added support of `waitForTransaction` for Solana network

### CODE REVIEW

File: src/context/WalletProvider.tsx [hunk 1]
```diff
@@ -587,20 +587,32 @@ const WalletProvider = function WalletProvider({ children }: { children: React.R
...
+    if (chainId === NETWORK_IDS.Solana) {
+      const cluster = getCluster(state.chainId)
+      const solanaNetwork = clusterApiUrl(cluster)
+      const connection = new Connection(solanaNetwork)
 
-    const tx = await state.provider.waitForTransaction(hash, confirmations)
-    if (tx.confirmations && tx.status !== REVERTED_STATUS) {
-      return tx
+      try {
+        await connection.getTransaction('12345678901234567890123456789012')
```
Comment (insulineru) [inline src/context/WalletProvider.tsx]: what is this number? an artefact after development?

File: src/types.ts [hunk 2]
```diff
@@ -46,7 +46,7 @@ type TWallet = {
   ) => Promise<string /* | false */> // todo: sendTx reject => false
   disconnect: () => void
   estimateGas: (data: TransactionRequest) => Promise<BigNumber | undefined>
-  waitForTransaction: (transactionHash: string, confirmations?: number) => Promise<TransactionReceipt>
+  waitForTransaction: (transactionHash: string, confirmations?: number) => Promise<void>
```
Comment (insulineru) [inline src/types.ts:49]: This is a change that fundamentally changes the logic of the function. We are often guided by the data we get from this function in router, so similar logic can be in the checkout. I would prefer to change the logic separately from restore Solana tx

**Review by insulineru — CHANGES_REQUESTED** [review]

File: src/context/WalletProvider.tsx (hunk 1 above)
Comment (nrjshka) [reply src/context/WalletProvider.tsx]: LOL, it's an old code:)

File: src/types.ts (hunk 2 above)
Comment (nrjshka) [reply src/types.ts:49]: We are not using result of `waitForTransaction` nowhere:) You can check it:)

Comment (nrjshka) [reply src/types.ts:49]: 1/ `useTransactionController`
image
2/ `HomePage`
image
3/ `ProgressPage`
image
4/ `useCheckTx`
image

**Review by insulineru — APPROVED** [review]

**Pull request merged**